core/consensus/protocols/protocols.go

```diff
@@ -0,0 +1,61 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package protocols
+
+import (
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+const (
+	protocolIDPrefix = "/charon/consensus/"
+
+	QBFTv2ProtocolID = "/charon/consensus/qbft/2.0.0"
+)
+
+// Protocols returns the supported protocols of this package in order of precedence.
+func Protocols() []protocol.ID {
+	return []protocol.ID{QBFTv2ProtocolID}
+}
+
+// MostPreferredConsensusProtocol returns the most preferred consensus protocol from the given list.
+func MostPreferredConsensusProtocol(protocols []string) string {
+	for _, p := range protocols {
+		if strings.HasPrefix(p, protocolIDPrefix) {
+			return p
+		}
+	}
+
+	return QBFTv2ProtocolID
+}
+
+// IsSupportedProtocolName returns true if the protocol name is supported.
+func IsSupportedProtocolName(name string) bool {
+	for _, p := range Protocols() {
+		nameAndVersion := strings.TrimPrefix(string(p), protocolIDPrefix)
```

Comment:
Wondering if this logic could be attached to the `protocol.ID` struct as a method.

Reply:
That was my original thought as well, but `protocol.ID` is the inner type of libp2p which we should not extend.. I could have introduce a derived type, but this seemed too much for this simple case.

Reply:
Ah right, no need to over-abstract then, let's keep it this way.